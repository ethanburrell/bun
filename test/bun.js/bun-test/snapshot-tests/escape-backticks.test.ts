import {
    describe,
    expect,
    it,
  } from "bun:test";
import { readFileSync } from "fs";

describe("test bun-test escapes backticks correctly", () => {
    it("`escape backticks`", () => {
        expect("escape the ` backtick!").toMatchSnapshot()
    })
})

describe("test format of snapshot", () => {
    it("snapshot file should be properly fomatted", () => {
        const data = readFileSync(import.meta.dir + "/__snapshots__/escape-backticks.test.ts.snap");
        
        const expected = [
            "// Jest Snapshot v1, https://goo.gl/fbAQLP",
            "exports[`test bun-test escapes backticks correctly \`escape backticks\` 1`] = `escape the \` backtick!`;",
            ""
        ]
        expected.forEach((v, i) => {
            expect(v).toBe(expected[i])
        })
    })
})
