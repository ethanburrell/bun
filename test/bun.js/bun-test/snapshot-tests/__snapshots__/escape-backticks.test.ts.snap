// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test bun-test escapes backticks correctly \`escape backticks\` 1`] = `escape the \` backtick!`;
